clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -519,6 +519,42 @@ public static String stackTrace(Throwable e) {
         return sw.toString();
     }
 
+    /**
+     * returns the root cause of a given throwable
+     * @param t a throwable
+     * @return the root cause of the argument
+     */
+    public static Throwable rootCause(Throwable t) {
+        Throwable cause = t;
+        while (cause != null && cause.getCause() != null && cause.getCause() != cause) {
+            cause = cause.getCause();
+        }
+        return cause;
+    }
+
+    /**
+     * returns a short human-readable description of a throwable
+     * @param t a throwable
+     * @return a concise description of the throwable
+     */
+    public static String describe(Throwable t) {
```

Comment:
Not sure of the name. `describe` suggests that it has done something, while it is actually just returning a description. Maybe `toString` or something along those lines?

Reply:
What is the motivation for printing the root cause without the intermediary exception messages? 

@cmccabe, `KafkaAdminClient` has a `prettyPrintException`. Is there a reason why it has to have its own method for that instead of a shared one like this one?

Reply:
`KafkaAdminClient#prettyPrintException` includes the class of the throwable in the string, which this method does not.  `KafkaAdminClient#prettyPrintException` is also intended to generate a short (one line) description, not dive into the stack traces and causes.  I'm sure there's some unification we could do at some point, though, if we had more utility methods for dealing with exception text

Reply:
@ijuma - (at least in my experience) the root cause tends to be the most useful exception. printing all of them would make it not fit in a single line, outer and inner-most seemed like a good compromise to me.
@cmccabe - this should still fit in a single line. we could also include the (simple) name of the exception classes and then re-use in both places?

Reply:
I don't think this is the right approach.  Even if the root cause is important, we still want to know that we experienced a disconnect.  The current code does not show this.

In general, we can't really make blanket statements like "the root cause is always the most useful thing"  It depends on how the code is using the exceptions.  Also, we shouldn't change what any of the exception output is without making a detailed examination of the before and after log output, and making sure that we like the "after" output better.

Reply:
current patch shows outermost and inner most exception. so you get root cause and you get to know that you experienced a disconnect

Reply:
Thanks for the clarification.  What's the rationale for printing just the outermost and innermost exception?  Perhaps we should print all of the exceptions, but simply skip the stack traces?